app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt

```diff
@@ -421,4 +432,27 @@ class CalculatorImpl(
     private fun Double.format() = formatter.doubleToString(this)
 
     private fun String.removeGroupSeparator() = formatter.removeGroupingSeparator(this)
+
+    fun getCalculatorStateJson(): JSONObject {
+        val jsonObj = JSONObject()
+        jsonObj.put(RES, currentResult)
+        jsonObj.put(PREVIOUS_CALCULATION, previousCalculation)
+        jsonObj.put(LAST_KEY, lastKey)
+        jsonObj.put(LAST_OPERATION, lastOperation)
+        jsonObj.put(BASE_VALUE, baseValue)
+        jsonObj.put(SECOND_VALUE, secondValue)
+        jsonObj.put(INPUT_DISPLAYED_FORMULA, inputDisplayedFormula)
+        return jsonObj
+    }
+
+    private fun setFromSaveInstanceState(json: String) {
+        val jsonObject = JSONTokener(json).nextValue() as JSONObject
+        currentResult = jsonObject.getString(RES)
+        previousCalculation = jsonObject.getString(PREVIOUS_CALCULATION)
+        baseValue = jsonObject.getDouble(BASE_VALUE)
+        secondValue = jsonObject.getDouble(SECOND_VALUE)
+        inputDisplayedFormula = jsonObject.getString(INPUT_DISPLAYED_FORMULA)
+        lastKey = jsonObject.getString(LAST_KEY)
+        lastOperation = jsonObject.getString(LAST_OPERATION)
```

Comment:
lets use the same order of constants at writing and reading json for better readability

Reply:
@tibbi  indeed it is better 👍 [da94377](https://github.com/SimpleMobileTools/Simple-Calculator/pull/305/commits/da943770df8da1502739fda30a941f33f155a3fc)